cmd/cueckoo/cmd/importpr.go

```diff
@@ -48,28 +55,84 @@ func importPRDef(c *Command, args []string) error {
 		return fmt.Errorf("%q is not a valid number", pr)
 	}
 
-	// TODO: validate that the number provided is indeed a valid PR
+	log.Printf("using github remote URL %q", cfg.githubURL)
 
-	// TODO: work out support for multiple PRs
+	branchName := fmt.Sprintf("importpr-%d", pr)
 
-	msg := fmt.Sprintf("Import PR %d from GitHub to Gerrit", pr)
-	payload, err := buildImportPRPayload(msg, importPRPayload{
-		PR: pr,
-	})
-	if err != nil {
+	// TODO: Note that mainErr's ctx is not wired here.
+	// We should wire it up and use it for e.g. FetchContext.
+	// For now, use a hard-coded timeout of 10s.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// If the branch already exists, refuse to continue.
+	if out, err := run(ctx,
+		"git", "show-ref", "--verify", "--quiet", fmt.Sprintf("refs/heads/%s", branchName),
+	); err == nil {
+		return fmt.Errorf("branch %q already exists; exiting. delete it to start over.", branchName)
+	} else if err != nil && len(out) > 0 {
+		return err // something else went wrong
+	}
+
+	// Fetch the PR HEAD and place it in a new branch, then switch to it.
+	if _, err := run(ctx,
+		"git", "fetch", "--quiet", cfg.githubURL,
+		fmt.Sprintf("pull/%d/head:%s", pr, branchName),
+	); err != nil {
 		return err
 	}
-	err = cfg.triggerRepositoryDispatch(cfg.githubOwner, cfg.githubRepo, payload)
-	if err != nil {
+	if _, err := run(ctx, "git", "switch", "--quiet", branchName); err != nil {
 		return err
 	}
-	return nil
-}
+	log.Printf("fetched PR into branch %q", branchName)
 
-type importPRPayload struct {
-	PR int `json:"pr"`
+	// Rebase all the commits on the remote tip, squashing all commits after the
+	// oldest one. We fetch the latest tip from the remote via HEAD, in case the
+	// local clone is not up to date.
+	if _, err := run(ctx, "git", "fetch", "--quiet", cfg.githubURL, "HEAD"); err != nil {
```

Comment:
TIL that this fetches the tip of the remote default branch.

Reply:
"default branch" is just the repo's HEAD :)